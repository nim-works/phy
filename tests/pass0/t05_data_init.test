discard """
  output: "(Done 64)"
"""
(TypeDefs
  (ProcTy (UInt 4)))
(GlobalDefs
  (GlobalDef (UInt 8) (Data 4 (StringVal "\x40\x00\x00\x00"))))
(ProcDefs
  (ProcDef (Type 0) 0 (Locals)
    (List
      (Block (Params)
        (Return
          (Load (UInt 4)
            (Copy (Global 0))))))))
