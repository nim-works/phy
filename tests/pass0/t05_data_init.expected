.type t0 () -> int
.global g0 int 0
.memory 4096
.init 0 "@\x00\x00\x00"
.reloc g0
.start t0 p0
  GetGlobal g0
  LdInt32 0
  Ret
.end
